packages/flame/lib/src/layout/align_component.dart

```diff
@@ -70,9 +70,6 @@ class AlignComponent extends PositionComponent {
   PositionComponent? get child => _child;
 
   set child(PositionComponent? value) {
-    if (_child == value) {
```

Comment:
```
  @mustCallSuper
  @override
  void onChildrenChanged(Component child, ChildrenChangeType type) {
    print('onChildrenChanged: $child, $type');
    if (_child?.parent != this) {
      this.child = null;
    }
  }
```
I have never seen a print of this method

Of course you need to understand the problem. But she is definitely not on the side of this component. Unfortunately, I don’t yet know all the intricacies of the framework. Perhaps the method is not called because we can add a child to the same parent.

Reply:
I would also like to add that these changes to the class will not affect performance, but will make this component work well.

Reply:
We should open a separate issue about this if it doesn't work, I'll merge this meanwhile.

Reply:
Agree